app/src/main/java/ru/alexzdns/fundamentals/homework/ui/moviesList/MoviesListViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package ru.alexzdns.fundamentals.homework.ui.moviesList
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import ru.alexzdns.fundamentals.homework.App
+import ru.alexzdns.fundamentals.homework.data.loadMovies
+import ru.alexzdns.fundamentals.homework.data.models.Movie
+
+class MoviesListViewModel : ViewModel() {
+    private val _mutableState = MutableLiveData<State>(State.Default())
+    val state: LiveData<State> get() = _mutableState
+
+    fun getMovies() {
+        viewModelScope.launch {
+            _mutableState.value = State.Loading()
+            try {
+                val movies = loadMovies(App.context())
```

Comment:
В текущей реализации класс получился не тестопригодным. Так как его невозможно подменить context реализацией для тестов. 
Чтобы класс был тестопрогодным, зависимости необходимо передавать через параметры конструктора.
```suggestion
                val movies = loadMovies(context)
```